### DISCUSSION TITLE
Snipe it installation

### DISCUSSION THREAD

**Discussion by Dol00** [snipe/snipe-it#15662] [General]
How to install snipe it in web server 2019 that act as operating system ?

**Comment by simisola88** [comment]
Do you mean windows server 2019?

**Reply by Dol00** [reply]
Yes

**Reply by simisola88** [reply]
[php-ini.txt](https://github.com/user-attachments/files/17512994/php-ini.txt)
Ensure your server meets the system requirements as specified here https://snipe-it.readme.io/docs/requirements#requirements-specific-to-windows-server

**Install IIS and Enable CGI** 
Install first the IIS from Server Manager . During IIS installation on Server Roles select Web Server (IIS), then Add Features and Next after selecting. At Role Services, expand Application Development and select [CGI (Common Gateway Interface).

**Enabling PHP on ISS**
Download PHP [https://windows.php.net/download/]
If you are using PHP as FastCGI with IIS you should use the Non-Thread Safe (NTS) versions of PHP.
- The VS16 and VS17 builds require to have the Visual C++ Redistributable for Visual Studio 2015-2022 [x64][https://aka.ms/vs/17/release/vc_redist.x64.exe] or [x86][https://aka.ms/vs/17/release/vc_redist.x86.exe] installed
Extract the downloaded ZIP file to a directory, e.g., C:\php.
Rename php.ini-development to php.ini
Open php.ini in a text editor and configure necessary settings (like enabling extensions). (I have attached my php.ini for reference
[php-ini.txt][https://github.com/user-attachments/files/17512947/php-ini.txt]


**Add PHP to Environment Variables**

    Right-click on "This PC" or "Computer" and select "Properties".
    Click on "Advanced system settings" and then "Environment Variables".
    Under "System variables", find the Path variable and click "Edit".
    Add the path to your PHP directory (e.g., C:\php).

**Configure IIS to Use PHP**
Add a Handler Mapping:

    In IIS Manager, select your server or site.
    In the Features view, double-click on "Handler Mappings".
    Click on "Add Module Mapping" on the right.
    Fill in the following:
        Request path: *.php
        Module: FastCgiModule
        Executable: C:\php\php-cgi.exe (or wherever your PHP is installed)
        Name: PHP_via_FastCGI

    Click "OK" and then confirm any prompts that appear.

**Create a PHP Info File**
    Create a file named info.php in the default site folder (usually C:\inetpub\wwwroot).
    Access the File:
    Open a web browser and go to http://localhost/info.php. You should see the PHP info page.

**Install MariaDB for windows**
**Create a DB, user and user-password for the DB** 
**Download composer**

Follow these instructions
[https://snipe-it.readme.io/docs/installation]
Follow these instruction to configure your IIS site [https://snipe-it.readme.io/docs/windowsiis](url)

**Reply by ericalbert8080-gif** [reply]
I figured it out.  I 
noticed that it said safe thread in the version, and I must have downloaded it instead of the non-safe thread version.  I reinstalled PHP and everything works now thx for your time
